inspirehep/modules/records/utils.py

```diff
@@ -44,3 +47,16 @@ def get_detailed_template_from_record(record):
     """Return the detailed template corresponding to the given record."""
     endpoint = get_endpoint_from_record(record)
     return current_app.config['RECORDS_UI_ENDPOINTS'][endpoint]['template']
+
+
+@contextmanager
+def file_resolving(fpath):
+    known_schemes = ['http', 'https']
+    if urlparse(fpath).scheme in known_schemes:
+        with closing(requests.get(url=fpath, stream=True)) as req:
+            if req.status_code == 200:
+                req.raw.decode_content = True
+                yield req.raw
+    else:
+        with open(fpath, mode='r') as fh:
```

Comment:
Here I would add a try: except: for `IOError` in case the path does not exist

Reply:
Do we need to silently continue if something is wrong with the path? @jacquerie